test/e2e/utils/helpers.go

```diff
@@ -680,3 +683,66 @@ func PodIsRunning(pod *corev1.Pod) (bool, error) {
 	}
 	return false, nil
 }
+
+func WaitUntilServingCertificateIsReloaded(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster) error {
+	servingCertSecret, err := client.Secrets(sc.Namespace).Get(ctx, fmt.Sprintf("%s-local-serving-certs", sc.Name), metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("can't get serving cert secret: %w", err)
+	}
+	servingCerts, err := ocrypto.DecodeCertificates(servingCertSecret.Data["tls.crt"])
+	if err != nil {
+		return fmt.Errorf("cam't decode serving certificate: %w", err)
+	}
+	if len(servingCerts) != 1 {
+		return fmt.Errorf("expected 1 serving certificate, got %d", len(servingCerts))
+	}
+	servingCert := servingCerts[0]
+
+	adminClientSecret, err := client.Secrets(sc.Namespace).Get(ctx, fmt.Sprintf("%s-local-user-admin", sc.Name), metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("can't get client certificate secret: %w", err)
+	}
+
+	adminTLSCert, err := tls.X509KeyPair(adminClientSecret.Data["tls.crt"], adminClientSecret.Data["tls.key"])
+	if err != nil {
+		return fmt.Errorf("can't parse client certificate: %w", err)
+	}
+
+	servingCABundleConfigMap, err := client.ConfigMaps(sc.Namespace).Get(ctx, fmt.Sprintf("%s-local-serving-ca", sc.Name), metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("can't get serving CA configmap: %w", err)
+	}
+
+	caBundleCerts, err := ocrypto.DecodeCertificates([]byte(servingCABundleConfigMap.Data["ca-bundle.crt"]))
+	if err != nil {
+		return fmt.Errorf("cam't decode serving CA certificate: %w", err)
+	}
+
+	servingCAPool := x509.NewCertPool()
+	for _, caCert := range caBundleCerts {
+		servingCAPool.AddCert(caCert)
+	}
+
+	hosts, err := GetHosts(ctx, client, sc)
+	if err != nil {
+		return fmt.Errorf("can't get v1.ScyllaCluster %q hosts: %w", naming.ObjRef(sc), err)
+	}
+
+	for _, host := range hosts {
+		o.Eventually(func(eo o.Gomega) {
+			serverCerts, err := GetServerTLSCertificates(fmt.Sprintf("%s:9142", host), &tls.Config{
+				ServerName:         host,
+				InsecureSkipVerify: false,
+				Certificates:       []tls.Certificate{adminTLSCert},
+				RootCAs:            servingCAPool,
+			})
+			eo.Expect(err).NotTo(o.HaveOccurred())
+			eo.Expect(serverCerts).NotTo(o.BeEmpty())
+			eo.Expect(serverCerts).To(o.HaveLen(1))
+
+			eo.Expect(serverCerts[0].Raw).To(o.Equal(servingCert.Raw))
```

Comment:
can you assert the raw ones are not empty?

Reply:
done